pkg/secrets/passphrase/manager.go

```diff
@@ -351,9 +353,27 @@ func readPassphrase(prompt string, useEnv bool) (phrase string, interactive bool
 		if phraseFile, ok := os.LookupEnv("PULUMI_CONFIG_PASSPHRASE_FILE"); ok && phraseFile != "" {
 			phraseFilePath, err := filepath.Abs(phraseFile)
 			if err != nil {
-				return "", false, fmt.Errorf("unable to construct a path the PULUMI_CONFIG_PASSPHRASE_FILE: %w", err)
+				return "", false, fmt.Errorf("unable to construct a path for PULUMI_CONFIG_PASSPHRASE_FILE: %w", err)
+			}
+			fileInfo, err := os.Stat(phraseFilePath)
+			if err != nil {
+				return "", false, fmt.Errorf("unable to stat PULUMI_CONFIG_PASSPHRASE_FILE=%q: %w", phraseFilePath, err)
+			}
+			var phraseDetails []byte
+			if fileInfo.Mode().IsRegular() && (fileInfo.Mode().Perm()&0o111 != 0) {
+				// File is executable. Execute it and read its output.
```

Comment:
Note Ansible themselves don't have this issue because they don't support running on Windows.

Reply:
Are you saying that you'd rather not enter the complications of supporting this on Windows, or that some kind of flag should control whether this execution should be attempted, and the Windows implementation would differ from the Linux one? What if I rely on POSIX functionality here? Windows supports POSIX.

Reply:
I think the idea of an executable file to run to get the passphrase is a good feature, and if we add it then we should also support that feature on windows.
We can't support that feature on windows if it works via looking at exec bits on the file because windows doesn't have that. So we need a different approach for windows, get to that in a sec.
In general I prefer the system to behave the same between windows and linux when possible, so if the feature can be done in way B that works on both we should _just_ do B, not B on windows and exec bits on linux.
The most obvious way I can think of to support this on windows is to add a second envvar that either turns on treating PULUMI_CONFIG_PASSPHRASE_FILE as a script, or just adding a whole new envvar PULUMI_CONFIG_PASSPHRASE_SCRIPT (naming maybe could be better) that if sets is used before _FILE and is a path that we just execute.